test-authenticator/src/main/java/io/curity/identityserver/plugin/test/authentication/TestBackchannelAuthenticatorHandler.java

```diff
@@ -60,34 +64,50 @@ public boolean startAuthentication(String authReqId,
     {
         _logger.trace("checkAuthentication() called.");
 
+        String subject = mutableRequestSubjectMap.get(authReqId).getKey();
+
         if (!mutableRequestSubjectMap.containsKey(authReqId))
         {
             return Optional.of(new BackchannelAuthenticationResult(null,
                     BackchannelAuthenticatorState.EXPIRED));
         }
-        else if ("denying-user".equals(mutableRequestSubjectMap.get(authReqId)))
+        else if ("denying-user".equals(subject))
         {
             return Optional.of(new BackchannelAuthenticationResult(null,
                     BackchannelAuthenticatorState.FAILED));
         }
 
-        if (!mutableRequestStateMap.containsKey(authReqId))
-        {
-            // for test purposes, first time return STARTED status and SUCCEEDED subsequently
-            mutableRequestStateMap.put(authReqId, BackchannelAuthenticatorState.STARTED);
-            _logger.trace("Authentication pending, still..");
-            return Optional.of(new BackchannelAuthenticationResult(null,
-                    BackchannelAuthenticatorState.STARTED));
-        }
-        else
+        if (mutableRequestStateMap.containsKey(authReqId))
         {
+            if (_backchannelConfiguration.getDelay().isPresent())
+            {
+                Instant startAuthInstant = mutableRequestSubjectMap.get(authReqId).getValue();
+                Instant continueInstant = startAuthInstant.plus(_backchannelConfiguration.getDelay().get(), SECONDS);
+                Instant now = Instant.now();
+
+                if (now.isBefore(continueInstant))
+                {
+                    _logger.trace("Authentication pending till {}..", continueInstant);
+
+                    return Optional.empty();
```

Comment:
I think you should return `STARTED` (i.e. the current status). Returning `empty` I think has the effect of telling the controller the status is not known.